backend/daemon/api/documents/v1alpha/documents.go

```diff
@@ -443,6 +458,82 @@ func (api *Server) DeletePublication(ctx context.Context, in *documents.DeletePu
 	return &emptypb.Empty{}, nil
 }
 
+// PushPublication implements the corresponding gRPC method.
+func (api *Server) PushPublication(ctx context.Context, in *documents.PushPublicationRequest) (*emptypb.Empty, error) {
+	if in.DocumentId == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "must specify publication ID")
+	}
+
+	if in.Version == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "must specify version")
+	}
+
+	// If no gwClient is set we can't do anything else.
+	if api.gwClient == nil {
+		return nil, status.Errorf(codes.FailedPrecondition, "there is no gwClient definition")
+	}
+
+	eid := hyper.EntityID(in.DocumentId)
+
+	heads, err := hyper.Version(in.Version).Parse()
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to parse version %s: %v", in.Version, err)
+	}
+
+	entity, err := api.blobs.LoadEntityFromHeads(ctx, eid, heads...)
+
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unable to get blobs from head: %v", err)
+	}
+
+	if entity == nil {
+		return nil, status.Errorf(codes.NotFound, "no published changes for entity %s", eid.String())
+	}
+
+	conn, cancelFcn, err := api.db.Conn(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unable to get db connection: %v", err)
+	}
+	defer cancelFcn()
+
+	gdb, err := hypersql.EntitiesLookupID(conn, entity.ID().String())
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "unable to find entity id [%s]: %v", entity.ID().String(), err)
+	}
+	if gdb.ResourcesID == 0 {
+		return nil, status.Errorf(codes.NotFound, "document %s not found", entity.ID().String())
+	}
+
+	cids := []cid.Cid{}
+	err = sqlitex.Exec(conn, groups.QCollectBlobs(), func(stmt *sqlite.Stmt) error {
+		var (
+			id        int64
+			codec     int64
+			multihash []byte
+		)
+		stmt.Scan(&id, &codec, &multihash)
+
+		c := cid.NewCidV1(uint64(codec), multihash)
+		cids = append(cids, c)
+		return nil
+	}, gdb.ResourcesID)
```

Comment:
There's a bit of a problem with this code. This query would collect *all* the blobs for a given resource (or entity). But here we do accept the specific version. So even if you've checked above that version exists and so on, we'll still push all the blobs. So if you want to push an earlier version of the document, you'll end up pushing all of the changes. Not sure if it's a big problem, but it's something to keep in mind.

Reply:
Worst case we will have old version no one care about in the gateway. No big deal wen we hace gc. Best case scenario, the user can navigate older versions which may be valuable. Should not impact in performance much...

Reply:
The problem is not about older version. It's about newer versions. Basically we could remove the `version` field from the request entirely, because it doesn't do anything, and we can only publish whole documents with all the changes.

Reply:
Yeah, maybe just do that for now. Over-pushing is generally fine.